clippy_lints/src/array_indexing.rs

```diff
@@ -61,11 +60,11 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for ArrayIndexing {
             // Array with known size can be checked statically
             let ty = cx.tables.expr_ty(array);
             if let ty::TyArray(_, size) = ty.sty {
-                let size = ConstInt::Infer(size as u128);
```

Comment:
Array lengths are usize, so you need `ConstInt::Usize(ConstUsize::new(size as u64, cx.sess().target.uint_type))`